plugins/chat/assets/javascripts/discourse/services/chat-channel-pane-subscriptions-manager.js

```diff
@@ -0,0 +1,189 @@
+import Service, { inject as service } from "@ember/service";
+import EmberObject from "@ember/object";
+import ChatMessage from "discourse/plugins/chat/discourse/models/chat-message";
+import { cloneJSON } from "discourse-common/lib/object";
+import { bind } from "discourse-common/utils/decorators";
+
+export default class ChatChannelPaneSubscriptionsManager extends Service {
```

Comment:
I think I could even extract a further utility base service out of this that has all this code in it, then any time we need a new subscription management service we can inherit from that.

```js
get messageBusChannel() {
    return `/chat/${this.model.id}`;
  }

  get messageBusLastId() {
    return this.model.channelMessageBusLastId;
  }

  get messagesManager() {
    return this.model.messagesManager;
  }

  subscribe(model) {
    this.unsubscribe();
    this.model = model;
    this.messageBus.subscribe(
      this.messageBusChannel,
      this.onMessage,
      this.messageBusLastId
    );
  }

  unsubscribe() {
    if (!this.model) {
      return;
    }
    this.messageBus.unsubscribe(this.messageBusChannel, this.onMessage);
    this.model = null;
  }
```